### ISSUE TITLE
Don't know how to parse column type: 28

### ISSUE THREAD

**Issue by kluzzebass** [denodrivers/postgres#117]
```
error: Uncaught Error: Don't know how to parse column type: 28
```

28 is the transaction identifier (xid), which is a "32-bit quantity", according to the docs, so I think it can be safely decoded as an int4. The same goes for 29, the command identifier (cid).

Obviously, this is related to #99.

**Comment by bartlomieju** [comment]
@kluzzebass I'm open to expanding the list for as many values as possible

**Comment by kluzzebass** [comment]
@bartlomieju I see what you did there. :)I'll see what I can do about it.

**Comment by hayd** [comment]
Is there a canonical list of these mappings (is there some prior art / code)?

**Comment by uki00a** [comment]
Hi, I've made a list of mappings based on [the node-pg-types's source](https://github.com/brianc/node-pg-types/blob/master/lib/textParsers.js) and [decode.ts](https://github.com/deno-postgres/deno-postgres/blob/master/encode.ts).

## Unmplemented types

|oid|type name|TS type|
|:---:|:---:|:---:|
|600|point|`{ x: number, y: number }`|
|718|circle|`{ x: number, y: number, radius: number }`|
|1000|_bool|`boolean[]`|
|1001|_bytea|`Buffer`|
|1028|oid[]|`number[]`|
|1016|_int8|`string[]`|
|1017|point[]|`{ x: number, y: number }[]`|
|1021|_float4|`number[]`|
|1022|_float8|`number[]`|
|1231|_numeric|`string[]`|
|1008|_regproc|`string[]`|
|1115|timestamp without time zone[]|`Date[]`|
|1182|_date|`Date[]`|
|1185|timestamp with time zone[]|`Date[]`|
|1186|interval|See https://github.com/bendrucker/postgres-interval/blob/master/index.d.ts|
|1187|_interval|See https://github.com/bendrucker/postgres-interval/blob/master/index.d.ts|
|199|json[]|`object[]`|
|3807|jsonb[]|`object[]`|
|3907|numrange[]|`string[]`|
|791|money[]|`string[]`|
|1183|time[]|`string[]`|
|1270|timetz[]|`string[]`|
|*|_unimplemented types other than the above_|`string`|

**Comment by Soremwar** [comment]
199 abd 3807 addressed on #179

**Comment by sanderhahn** [comment]
The whole list of type oids is dynamic and is available from [pg_catalog.pg_type](https://www.postgresql.org/docs/current/catalog-pg-type.html). The list also contains information about arrays and elements. Some actions can also create new type oids:

```sql
CREATE TYPE compfoo AS (f1 int, f2 text);

select (1,2)::compfoo;
--   row  
-- -------
--  (1,2)
-- (1 row)

select pg_typeof((1,2)::compfoo)::regtype::oid;
--  pg_typeof 
-- -----------
--     721528
-- (1 row)
```

Maybe its possible to query this table at startup time so that the driver knows how to convert values?
